src/main/java/com/github/jcustenborder/kafka/connect/spooldir/InputFile.java

```diff
@@ -197,7 +210,50 @@ public long lastModified() {
     return this.lastModified;
   }
 
+  private List<File> getInputPathSubDirsToCleanup() {
+    List<File> inputPathSubDirsToCleanup = null;
+    if (this.inputPathSubDir != null && !config.inputPathWalkRecursivelyRetainSubDirs) {
+      inputPathSubDirsToCleanup = new ArrayList<File>();
+      File lastSubDir = this.config.inputPath;
+      for (String subDirName : this.inputPathSubDir.split(File.separator)) {
+        lastSubDir = new File(lastSubDir, subDirName);
+        inputPathSubDirsToCleanup.add(lastSubDir);
+      }
+      Collections.reverse(inputPathSubDirsToCleanup);
+    }
+    return inputPathSubDirsToCleanup;
+  }
+
+  private void cleanupInputDirSubDirs() {
+    List<File> inputPathSubDirsToCleanup = this.getInputPathSubDirsToCleanup();
+    if (inputPathSubDirsToCleanup != null) {
+      for (File subDir : inputPathSubDirsToCleanup) {
+        try {
+          if (subDir.listFiles() == null || subDir.listFiles().length == 0) {
+            if (!subDir.delete()) {
+              log.error("Failed to delete input.path sub-directory: {}", subDir);
+            } else {
+              log.info("Cleaned up input.path sub-directory: {}", subDir);
+            }
+          } else {
+            log.info("Cannot clean up input.path sub-directory as it is not empty: {}", subDir);
+          }
+        } catch (SecurityException e) {
+          log.error("SecurityException thrown while trying to delete input.path sub-directory: {}", subDir, e);
+        }
+      }
+    }
+  }
+
+
+
   public void moveToDirectory(File outputDirectory) {
+
+    if (this.inputPathSubDir != null) {
+      outputDirectory = new File(outputDirectory, this.inputPathSubDir);
+      outputDirectory.mkdirs();
```

Comment:
Should we do a check that the directory exists before the mkdirs() call?

Reply:
I don't think its necessary as `mkdirs()` takes care of this. `inputPathSubDir` is unknown so you want to crawl the path it represents doing an exists() on each path item and a separate `mkdir()`?

Reply:
I was just thinking `if(!outputDirectory.isDirectory())` then call mkdirs().

Reply:
@jcustenborder this is checked in